CustomLookup/CustomEntitySearch.cs

```diff
@@ -0,0 +1,187 @@
+            }
+
+            public string RecordId { get; set; }
+            public InputRecordData Data { get; set; }
+        }
+
+        private class WebApiRequest
+        {
+            public List<InputRecord> Values { get; set; }
+        }
+        #endregion
+
+        #region Classes used to serialize the response
+
+        private class OutputRecord
+        {
+            public class OutputRecordData
+            {
+                public string Name { get; set; } = "";
+                public string MatchIndex { get; set; } = "";
+            }
+
+            public string RecordId { get; set; }
+            public List<OutputRecordData> Data { get; set; }
+        }
+
+        private class WebApiResponse
+        {
+            public List<OutputRecord> Values { get; set; }
+        }
+        #endregion
+
+        #region The Azure Function definition (JSON conversion, schema check, and response production all from sample)
+
+        [FunctionName("CustomEntitySearch")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            log.LogInformation("Custom Entity Search function: C# HTTP trigger function processed a request.");
+
+            var response = new WebApiResponse
+            {
+                Values = new List<OutputRecord>()
+            };
+
+            string requestBody = new StreamReader(req.Body).ReadToEnd();
+            var data = JsonConvert.DeserializeObject<WebApiRequest>(requestBody);
+                              
+            // Do some schema validation
+            if (data == null)
+            {
+                return new BadRequestObjectResult("The request schema does not match expected schema.");
+            }
+            if (data.Values == null)
+            {
+                return new BadRequestObjectResult("The request schema does not match expected schema. Could not find values array.");
+            }
+
+            // Calculate the response for each value.
+            foreach (var record in data.Values)
+            {
+                if (record == null || record.RecordId == null) continue;
+
+                if (record.Data.Text == null)
+                {
+                    return new BadRequestObjectResult("The request schema does not match expected schema. Could not find text string.");
+                }
+
+                if (record.Data.Words == null)
+                {
+                    return new BadRequestObjectResult("The request schema does not match expected schema. Could not find words array.");
+                }
+
+                OutputRecord responseRecord = new OutputRecord
+                {
+                    RecordId = record.RecordId
+                };
+
+                try
+                {
+                    responseRecord.Data = await FindEntity(record.Data.Text, record.Data.Words);
```

Comment:
Once you refactor to use the utility classes, this should become the heart of this method, with little plumbing around it. Look at other skills in the repo for reference.